gems/quilt_rails/lib/quilt_rails/performance/reportable.rb

```diff
@@ -4,8 +4,12 @@ module Quilt
   module Performance
     module Reportable
       def process_report(&block)
-        params.transform_keys! { |key| key.underscore.to_sym }
-        Client.send!(Report.from_params(params), &block)
+        report_params = if request.content_type == 'text/plain'
+          ActionController::Parameters.new(JSON.parse(request.body.read))
```

Comment:
I think the more "rubyist" way to do this would be to pull the massaging here out into it's own private method that looks something like:

```ruby
def normalized_params
  return params unless request.content_type == 'text/plain'
  
  ActionController::Parameters.new(JSON.parse(request.body.read))
end
```

and then use that in the `Client.send` here. Personally, I'm about even on either implementation, but I think what I suggested would be more idiomatic to ruby/rails